plugins/woocommerce/src/Internal/Orders/OrderActionsRestController.php

```diff
@@ -0,0 +1,128 @@
+ */
+class OrderActionsRestController extends RestApiControllerBase {
+	use AccessiblePrivateMethods;
+
+	/**
+	 * Get the WooCommerce REST API namespace for the class.
+	 *
+	 * @return string
+	 */
+	protected function get_rest_api_namespace(): string {
+		return 'order-actions';
+	}
+
+	/**
+	 * Register the REST API endpoints handled by this controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->route_namespace,
+			'/orders/(?P<id>[\d]+)/actions/send_order_details',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => fn( $request ) => $this->run( $request, 'send_order_details' ),
+					'permission_callback' => fn( $request ) => $this->check_permissions( $request ),
+					'args'                => $this->get_args_for_order_actions(),
+					'schema'              => $this->get_schema_for_order_actions(),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Permission check for REST API endpoint.
+	 *
+	 * @param WP_REST_Request $request The request for which the permission is checked.
+	 * @return bool|WP_Error True if the current user has the capability, otherwise a WP_Error object.
+	 */
+	private function check_permissions( WP_REST_Request $request ) {
+		$order_id = $request->get_param( 'id' );
+		$order    = wc_get_order( $order_id );
+
+		if ( ! $order ) {
+			return new WP_Error( 'woocommerce_rest_not_found', __( 'Order not found', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		return $this->check_permission( $request, 'read_shop_order', $order_id );
+	}
+
+	/**
+	 * Get the accepted arguments for the POST request.
+	 *
+	 * @return array[]
+	 */
+	private function get_args_for_order_actions(): array {
+		return array(
+			'id' => array(
+				'description' => __( 'Unique identifier of the order.', 'woocommerce' ),
+				'type'        => 'integer',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		);
+	}
+
+	/**
+	 * Get the schema for both the GET and the POST requests.
+	 *
+	 * @return array[]
+	 */
+	private function get_schema_for_order_actions(): array {
+		$schema['properties'] = array(
+			'message' => array(
+				'description' => __( 'A message indication whether the email was sent.', 'woocommerce' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		);
+		return $schema;
+	}
+
+	/**
+	 * Handle the POST /orders/{id}/actions/send_order_details.
+	 *
+	 * @param WP_REST_Request $request The received request.
+	 * @return array|WP_Error Request response or an error.
+	 */
+	public function send_order_details( WP_REST_Request $request ) {
+		$order_id = $request->get_param( 'id' );
+		$order    = wc_get_order( $order_id );
+		if ( ! $order ) {
+			return new WP_Error( 'invalid_order', __( 'Invalid order ID.', 'woocommerce' ), array( 'status' => 404 ) );
```

Comment:
Seems like the error message here and the one up in `check_permissions` should be the same, since they are for the same problem. What about using this for both:

```suggestion
			return new WP_Error( 'woocommerce_rest_invalid_order', __( 'Invalid order ID.', 'woocommerce' ), array( 'status' => 404 ) );
```

Reply:
Fixed!